### PR TITLE
bump pytables to 3.5.1 #24839

### LINKED ISSUE #24839: ValueError: cannot set WRITEABLE flag to True of this array
will need to revert the xfail decorator in:  https://github.com/pandas-dev/pandas/pull/25517 when this is fixed

#### Code Sample, a copy-pastable example if possible
Im getting all of a sudden this Error, any idea?

```python
# Your code here
 input_df = pd.read_hdf(path_or_buf='x.hdf5',key='/x',mode='r')
```
#### Problem description
Traceback :
````
Traceback (most recent call last):
  File "...", line 115, in <module>
    input_df = pd.read_hdf(path_or_buf='x.hdf5',key='/x',mode='r')
  File "/usr/local/lib/python3.6/dist-packages/pandas/io/pytables.py", line 394, in read_hdf
    return store.select(key, auto_close=auto_close, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pandas/io/pytables.py", line 741, in select
    return it.get_result()
  File "/usr/local/lib/python3.6/dist-packages/pandas/io/pytables.py", line 1483, in get_result
    results = self.func(self.start, self.stop, where)
  File "/usr/local/lib/python3.6/dist-packages/pandas/io/pytables.py", line 734, in func
    columns=columns)
  File "/usr/local/lib/python3.6/dist-packages/pandas/io/pytables.py", line 2937, in read
    start=_start, stop=_stop)
  File "/usr/local/lib/python3.6/dist-packages/pandas/io/pytables.py", line 2489, in read_array
    ret = node[0][start:stop]
  File "/usr/local/lib/python3.6/dist-packages/tables/vlarray.py", line 681, in __getitem__
    return self.read(start, stop, step)[0]
  File "/usr/local/lib/python3.6/dist-packages/tables/vlarray.py", line 821, in read
    listarr = self._read_array(start, stop, step)
  File "tables/hdf5extension.pyx", line 2155, in tables.hdf5extension.VLArray._read_array
ValueError: cannot set WRITEABLE flag to True of this array
```

### PR DESCRIPTION
**Pull request by fangchenli**
- [x] closes #24839
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

PyTables 3.5.1 was released on March 14, 2019. And it offers better support for np16.

### CODE REVIEW

File: pandas/tests/io/pytables/test_store.py [hunk 1]
```diff
@@ -2164,7 +2160,6 @@ def test_unimplemented_dtypes_table_columns(self, setup_path):
             with pytest.raises(TypeError):
                 store.append("df_unimplemented", df)
 
-    @td.xfail_non_writeable
```
Comment (jreback) [inline pandas/tests/io/pytables/test_store.py:2167]: can remove these version requirements as they are obsolete

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by WillAyd — APPROVED** [review]
lgtm

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
can you add a whatsnew bug fix note (I/O) that the issue is fixed in the OP; can merge after that.

**Comment by jreback** [comment]
thanks @fangchenli

**Pull request merged**